packages/components/src/scroll-lock/index.js

```diff
@@ -1,114 +1,64 @@
+import { useEffect } from '@wordpress/element';
+
+let previousScrollTop = 0;
 
 /**
- * Creates a ScrollLock component bound to the specified document.
- *
- * This function creates a ScrollLock component for the specified document
- * and is exposed so we can create an isolated component for unit testing.
- *
- * @param {Object} args Keyword args.
- * @param {HTMLDocument} args.htmlDocument The document to lock the scroll for.
- * @param {string} args.className The name of the class used to lock scrolling.
- * @return {WPComponent} The bound ScrollLock component.
+ * @param {boolean} locked
+ * @param {string} className
  */
-export function createScrollLockComponent( {
-	htmlDocument,
-	className = 'lockscroll',
-} = {} ) {
-	if ( ! htmlDocument && typeof document !== 'undefined' ) {
-		htmlDocument = document;
-	}
-
-	let lockCounter = 0;
-
-	/*
-	 * Setting `overflow: hidden` on html and body elements resets body scroll in iOS.
-	 * Save scroll top so we can restore it after locking scroll.
-	 *
-	 * NOTE: It would be cleaner and possibly safer to find a localized solution such
-	 * as preventing default on certain touchmove events.
-	 */
-	let previousScrollTop = 0;
-
-	/**
-	 * Locks and unlocks scroll depending on the boolean argument.
-	 *
-	 * @param {boolean} locked Whether or not scroll should be locked.
-	 */
-	function setLocked( locked ) {
-		const scrollingElement =
-			htmlDocument.scrollingElement || htmlDocument.body;
+function setLocked( locked, className ) {
+	const scrollingElement = document.scrollingElement || document.body;
 
-		if ( locked ) {
-			previousScrollTop = scrollingElement.scrollTop;
-		}
+	if ( locked ) {
+		previousScrollTop = scrollingElement.scrollTop;
+	}
 
-		const methodName = locked ? 'add' : 'remove';
-		scrollingElement.classList[ methodName ]( className );
+	const methodName = locked ? 'add' : 'remove';
+	scrollingElement.classList[ methodName ]( className );
 
-		// Adding the class to the document element seems to be necessary in iOS.
-		htmlDocument.documentElement.classList[ methodName ]( className );
+	// Adding the class to the document element seems to be necessary in iOS.
+	document.documentElement.classList[ methodName ]( className );
 
-		if ( ! locked ) {
-			scrollingElement.scrollTop = previousScrollTop;
-		}
+	if ( ! locked ) {
+		scrollingElement.scrollTop = previousScrollTop;
 	}
+}
 
-	/**
-	 * Requests scroll lock.
-	 *
-	 * This function tracks requests for scroll lock. It locks scroll on the first
-	 * request and counts each request so `releaseLock` can unlock scroll when
-	 * all requests have been released.
-	 */
-	function requestLock() {
+/*
+ * Setting `overflow: hidden` on html and body elements resets body scroll in iOS.
+ * Save scroll top so we can restore it after locking scroll.
+ *
+ * NOTE: It would be cleaner and possibly safer to find a localized solution such
+ * as preventing default on certain touchmove events.
+ */
```

Comment:
This looks like a comment for the `lockCounter` variable, but the original commented on `previousScrollTop`.